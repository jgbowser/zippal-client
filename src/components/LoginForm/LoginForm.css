input:focus {
  border: 3px solid #555;
  background-color: burlywood;
}
input {
  max-width: -webkit-fill-available;
}
legend {
  align-self: center;
  display: flex;
  flex-direction: column;
  padding: 100px 18px;
  max-width: fit-content;
  background-image: linear-gradient(360deg, white, darkkhaki);
  box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2), 0 6px 20px 0 rgba(0,0,0,0.19) ;
}
.user_form {
 align-self: center;
 text-align: -webkit-center;
}
.loginForm {
  display: flex;
}

button {
  margin-top: 4px;
  place-self: center;
}

#login-password-input, #login-username-input {
  align-self: center;
  max-width: -webkit-fill-available;
}

label{
  display: flex;
  justify-content: center;
}