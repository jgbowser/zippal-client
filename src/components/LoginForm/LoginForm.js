import React, {Component} from 'react'
import AuthApiService from '../../services/auth-api-service'
import UserContext from '../../context/UserContext'

import './LoginForm.css'

class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => { }
  }

  static contextType = UserContext

  state = {error: null}

  firstInput = React.createRef()

  handleSubmit = ev => {
    ev.preventDefault()
    this.props.toggleLoading()
    const {username, password} = ev.target

    this.setState({error:null})

    AuthApiService.postLogin({
      username:username.value,
      password:password.value,
    })
    .then(res => {
      username.value = ''
      password.value = ''
      this.context.processLogin(res.authToken)
      this.props.toggleLoading()
      this.props.onLoginSuccess()
    })
    .catch(res => {
      this.setState({ error: res.error})
      this.props.toggleLoading()
    })
  }

  render(){
    const { error } = this.state
    return (
      <form
        className='user_form'
        onSubmit={this.handleSubmit}
      >
       <legend>
        <div role='alert'>
          {error && <p className='error'>{error}</p>}
        </div>
        <label htmlFor='login-username-input'>
          Username
        </label>
        <input
          // ref={this.firstInput}
          id='login-username-input'
          name='username'
          required
        />
        <div>
          <label htmlFor='login-password-input'>
            Password
          </label>
          <input
            id='login-password-input'
            name='password'
            type='password'
            required
          />
        </div>
        <button className='form_button' type='submit'>
          Login
        </button>
        </legend>
      </form>
    )
  }
}

export default LoginForm