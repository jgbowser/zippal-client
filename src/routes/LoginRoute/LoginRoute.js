import React, {Component} from 'react'
import LoginForm from '../../components/LoginForm/LoginForm'
import Loading from '../../components/Loading/Loading'


class LoginRoute extends Component {
  constructor(props) {
    super(props)
    this.state = {
      loading: false
    }
  }
  static defaultProps = {
    location: {},
    history: {
      push: () => { },
    },
  }

  toggleLoading = () => {
    this.setState({
      loading: !this.state.loading
    })
  }


  handleLoginSuccess = () => {
    const {location, history} = this.props
    const destination = (location.state || {}).from || '/dashboard'
    history.push(destination)
  }

  render() {
    return (
      <section className='LoginSection'>
        <h2>Login</h2>
        {this.state.loading && <Loading />}
        <LoginForm
          onLoginSuccess={this.handleLoginSuccess}
          toggleLoading={this.toggleLoading}
        />
      </section>
    );
  }
}

export default LoginRoute