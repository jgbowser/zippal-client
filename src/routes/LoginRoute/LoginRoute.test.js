import React from 'react'
import ReactDOM from 'react-dom'
import LoginRoute from './LoginRoute'
import LoginForm from '../../components/LoginForm/LoginForm'

describe('LoginRoute Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<LoginForm />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})