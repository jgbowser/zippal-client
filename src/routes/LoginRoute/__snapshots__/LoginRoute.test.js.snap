// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginRoute Component takes snapshot of LoginRoute Component 1`] = `
<section
  className="LoginSection"
>
  <h2>
    Login
  </h2>
  <form
    className="user_form"
    onSubmit={[Function]}
  >
    <legend>
      <div
        role="alert"
      />
      <label
        htmlFor="login-username-input"
      >
        Username
      </label>
      <input
        id="login-username-input"
        name="username"
        required={true}
      />
      <div>
        <label
          htmlFor="login-password-input"
        >
          Password
        </label>
        <input
          id="login-password-input"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <button
        type="submit"
      >
        Login
      </button>
    </legend>
  </form>
</section>
`;
